code/datums/components/_component.dm

```diff
@@ -390,6 +401,15 @@
 						old_comp.InheritComponent(null, TRUE, arguments)
 					else
 						old_comp.InheritComponent(new_comp, TRUE)
+				if(COMPONENT_DUPE_SELECTIVE)
+					if(!new_comp)
+						new_comp = new nt(arglist(args))

```

Comment:
You should only make the new component if it fails the checks on the existing components